proto/src/determined/api/v1/api.proto

```diff
@@ -279,6 +279,27 @@ service Determined {
       tags: "Cluster"
     };
   }
+  // Post new maintenance messages.
+  rpc PostMaintenanceMessage(PostMaintenanceMessageRequest)
+      returns (PostMaintenanceMessageResponse) {
+    option (google.api.http) = {
+      patch: "/api/v1/master/maintenance_message"
```

Comment:
This feels like a PUT to me: we have one entity identified by `/api/v1/master/maintenance_message` which gets overwritten when somebody sets a new one, and people do DELETE to that same URI to unset it.